lib/rules/no-restricted-exports.js

```diff
@@ -63,6 +114,44 @@ module.exports = {
                     messageId: "restrictedNamed",
                     data: { name }
                 });
+                return;
+            }
+
+            if (name === "default") {
+                if (node.parent.type === "ExportAllDeclaration") {
+                    const isSourceSpecified = node.parent.source;
+
+                    if (isSourceSpecified && restrictDefaultExports && restrictDefaultExports.namespaceFrom) {
+                        context.report({
+                            node,
+                            messageId: "restrictedDefault"
+                        });
+                    }
+
+                } else { // ExportSpecifier
+                    const isSourceSpecified = node.parent.parent.source;
```

Comment:
```suggestion
                    const isSourceSpecified = !!node.parent.parent.source;
```

The variable name sounds like it should have a boolean value.